src/utils/mashupDocumentParser.ts

```diff
@@ -150,3 +159,93 @@ export const editSingleQueryMetadata = (metadataArray: Uint8Array, metadata: Met
+            endBuffer
+        );
+        
+        return newMetadataArray;
+    };
+
+    const updateConnectionOnlyMetadataStr = (metadataString: string, connectionOnlyQueryNames: string[]) => {
+        const parser: DOMParser = new DOMParser();
+        let metadataDoc: Document = parser.parseFromString(metadataString, xmlTextResultType);      
+        connectionOnlyQueryNames.forEach((queryName: string) => {
+            const items: Element = metadataDoc.getElementsByTagName(element.items)[0];
+            const stableEntriesItem: Element = createStableEntriesItem(metadataDoc, queryName);
+            items.appendChild(stableEntriesItem);
+            const sourceItem: Element = metadataDoc.createElementNS(metadataDoc.documentElement.namespaceURI, element.item);
+            sourceItem.appendChild(createItemLocation(metadataDoc, queryName, true));
+            const stableEntries: Element = metadataDoc.createElementNS(metadataDoc.documentElement.namespaceURI, element.stableEntries);
+            sourceItem.appendChild(stableEntries);
+            items.appendChild(sourceItem);
+        });    
+
+        const updatedMetdataString: string = new XMLSerializer().serializeToString(metadataDoc);
+
+        return updatedMetdataString;
+    };
+
+    const createItemLocation = (metadataDoc: Document, queryName: string, isSource: boolean) => {
+        const newItemLocation: Element = metadataDoc.createElementNS(metadataDoc.documentElement.namespaceURI, element.itemLocation);
+        const newItemType: Element = metadataDoc.createElementNS(metadataDoc.documentElement.namespaceURI, element.itemType);
+        newItemType.textContent = "Formula";
+        const newItemPath: Element = metadataDoc.createElementNS(metadataDoc.documentElement.namespaceURI, element.itemPath);
+        newItemPath.textContent = itemPathTextContext(queryName, isSource);
+        newItemLocation.appendChild(newItemType);
+        newItemLocation.appendChild(newItemPath);        
+        
+        return newItemLocation;
+    };
+
+    const createStableEntriesItem = (metadataDoc: Document, queryName: string) => {
+        const newItem: Element = metadataDoc.createElementNS(metadataDoc.documentElement.namespaceURI, element.item);
+        newItem.appendChild(createItemLocation(metadataDoc, queryName, false));
+        const stableEntries: Element = createConnectionOnlyEntries(metadataDoc);
+        newItem.appendChild(stableEntries);
+        
+        return newItem;
+    };
+
+    const createElementObject = (metadataDoc: Document, type: string, value: string) => {
+        const elementObject: Element = metadataDoc.createElementNS(metadataDoc.documentElement.namespaceURI, element.entry);
+        elementObject.setAttribute(elementAttributes.type, type);
+        elementObject.setAttribute(elementAttributes.value, value);
+        
+        return elementObject;
+    };
+    
+    const createConnectionOnlyEntries = (metadataDoc: Document) => {
```

Comment:
Let's generalize this method to create any kind of query,  createQueryEntries(fillObjectType (can be enum), property bug for the extra data according to the fillObject) for now will work only for connection only